examples/acm-terraform-blog/part1/README.md

```diff
@@ -0,0 +1,77 @@
+# Enable ACM features with Terraform
+
+## Part 1
+
+1. Clone this repo
+1. Set variables that will be used in multiple commands:
+
+    ```bash
+    FOLDER_ID = [FOLDER]
+    BILLING_ACCOUNT = [BILLING_ACCOUNT]
+    PROJECT_ID = [PROJECT_ID]
+    ```
+
+1. Create project:
+
+    ```bash
+    gcloud auth login
+    gcloud projects create $PROJECT_ID --name=$PROJECT_ID --folder=$FOLDER_ID
+    gcloud alpha billing projects link $PROJECT_ID --billing-account $BILLING_ACCOUNT
+    gcloud config set project $PROJECT_ID
+    ```
+
+1. Enable the Google Cloud APIs that will be used for this example and also enable the new configuration management feature for your project:
+
+    ```bash
+    gcloud services enable --project $PROJECT_ID  container.googleapis.com                \
+                                                    gkehub.googleapis.com                 \
+                                                    anthosconfigmanagement.googleapis.com
```

Comment:
Since this is an e2e example, why not enable in TF like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/35b5d5644340141abd546e13e34858bfd04b5941/examples/safer_cluster_iap_bastion/apis.tf

Reply:
This is example will accompany the blog intended for a broad audience, so we are trying to optimize for clarity and simplicity over re-usability. Hence we would like not to use additional modules if possible and stay with basic concepts.

Reply:
I'm not sure that really follows. For a general audience, don't we want the simplest configuration with the fewest manual steps?

Reply:
The intention TF file is kept as minimal as possible while enabling ACM features, so only a basic knowledge of Terraform provider functionality is needed, not larger ecosystem knowledge. It is not a big difference, but adding a module adds to the cognitive load. 
At same time I'd like to make sure that we don't bring something that is not consistent with the guidelines of your repo under your maintenance.  Let me know if you feel that using a module for enablement and a module for GKE is part of the requirements for being part of this examples.

Reply:
I'm not convinced it significantly increases the cognitive load to include this. If anything, I think it's simpler to avoid manual steps.

We also have a number of issues with the bare `google_container_cluster` resource when trying to improve over time—it manages node pools in-line, which is generally an antipattern.

One suggestion I would have is to split out the main.tf (where the actual ACM parts go) and gke.tf (where you put dependencies / not the main focus of the tutorial).